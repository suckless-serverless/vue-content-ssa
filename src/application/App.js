import Vue from 'vue'
import VueRouter from 'vue-router'
import { Application } from 'suckless-serverless'
import App from './App.vue'
import Home from './components/Pages/Home.vue'
import About from './components/Pages/About.vue'
import News from './components/Pages/News.vue'
import Contacts from './components/Pages/Contacts.vue'

Vue.use(VueRouter)

export default class VueApplication extends Application {
  handle (root) {
    const routes = [
      { path: '/', component: Home },
      { path: '/home', component: Home },
      { path: '/news', component: News },
      { path: '/contacts', component: Contacts },
      { path: '/about', component: About }
    ]

    const router = new VueRouter({
      routes // short for `routes: routes`
    })
    return new Promise((resolve, reject) => {
      new Vue({
        router,
        render: h => h(App)
      }).$mount('#' + root)
    })
  }
}
