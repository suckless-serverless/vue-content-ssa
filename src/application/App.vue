<template>
  <RouterView />
</template>

<script>

export default {
  name: 'App',
  data () {
    return {
      name: 'Number',
      number: 0
    }
  },
  computed: {
  },
  methods: {
  }
}
</script>

<style>
#app {
  font-family: "Avenir", Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
.full-width{
  border-width: medium;
  border-color: red;
}
</style>
