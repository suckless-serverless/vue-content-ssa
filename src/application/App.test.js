import { mount } from '@vue/test-utils'
import App from './App.vue'

describe('App.test.js', () => {
  let wrapper, vm

  beforeEach(() => {
    wrapper = mount(App, {
      data: () => ({
        // Replace data value with this fake data
        number: 1
      })
    })
    vm = wrapper.vm
  })

  it('equals number to be 1', () => {
    expect(vm.number).toEqual(1)
  })

  it('has the expected html structure', () => {
    expect(vm.$el).toMatchSnapshot()
  })

  it('click event', () => {
    const container = wrapper.find('button')
    container.trigger('click')
    expect(vm.$el).toMatchSnapshot()
  })
})
