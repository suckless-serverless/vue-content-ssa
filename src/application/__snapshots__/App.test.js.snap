// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.test.js click event 1`] = `
<div>
  <img
    alt="Vue logo"
    class=""
    height="200px"
    src="https://upload.wikimedia.org/wikipedia/commons/thumb/9/95/Vue.js_Logo_2.svg/1200px-Vue.js_Logo_2.svg.png"
  />
   
  <button>
    Click Me
  </button>
  
  Number of clicks =&gt; 2
  
  <!---->
</div>
`;

exports[`App.test.js has the expected html structure 1`] = `
<div>
  <img
    alt="Vue logo"
    class=""
    height="200px"
    src="https://upload.wikimedia.org/wikipedia/commons/thumb/9/95/Vue.js_Logo_2.svg/1200px-Vue.js_Logo_2.svg.png"
  />
   
  <button>
    Click Me
  </button>
  
  Number of clicks =&gt; 1
  
  <!---->
</div>
`;
